test/parallel/test-http2-compat-serverresponse-statusmessage.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const h2 = require('http2');
+
+// Http2ServerResponse.writeHead should accept an optional status message
+
+const server = h2.createServer();
+server.listen(0, common.mustCall(function() {
+  const port = server.address().port;
+  server.once('request', common.mustCall(function(request, response) {
+    const statusCode = 200;
+    const statusMessage = 'OK';
+    const headers = {'foo-bar': 'abc123'};
+    common.expectWarning(
```

Comment:
`process.emitWarning` does:

```js
function doEmitWarning(warning) {
  return function() {
    process.emit('warning', warning);
  };
}
// ...
process.emitWarning = function(warning, type, code, ctor) {
  // ...
  process.nextTick(doEmitWarning(warning));
};
```
... which is why we'll always win that race condition. 😫

And `common.expectWarning` does support a map (type/description) to check multiple warnings, but it seems messy to include the second warning in this assertion. (Removing/editing the `lib/http2.js` warning shouldn't break this statusMessage test.)

So I'll add a custom handler, thanks for the suggestion!